packages/edit-site/src/components/header/index.js

```diff
@@ -21,6 +22,7 @@ export default function Header() {
 			</h1>
 			<div className="edit-site-header__actions">
 				<SaveButton />
+				<PluginSidebar.PinnedItemsSlot scope="edit-site/sidebar" />
```

Comment:
I don't think that `PinnedItems` should be specific to the sidebar. It's a completely independent feature that provides shortcuts to access plugins. The current implementation might be misleading:
https://github.com/WordPress/gutenberg/blob/ec68bb8ddfca80ee4ce193408941fb0a49f8a6ee/packages/edit-post/src/components/sidebar/plugin-sidebar/index.js#L34-L46

It's colocated with the Sidebar component that has this button that allows to pin or unpin it. However, you could easily make the menu item that executes an action pinnable. Everything can be pinnable. There is probably a lot of prior knowledge hidden on GitHub accumulated in the discussions between me and @jasmussen.

Reply:
Thank you for the ping. Yes, for one, any item that has a pinned button in the toolbar _must have an item in the more menu_. Otherwise you can unpin that item and not get it back. I believe this is currently something the developer must ensure, but it would be nice if the API handled it so that a pinned item is an extension of a menu item.

And also, the pinned items do not need to have anything to do with sidebars, it could be sidebars, but it could be a popover, or a modal dialog, or hey, a link that opens a new tab. Think of it very much like a Chrome or Firefox extension in that way.

Reply:
@gziolo Are you saying it's _currently_ possible to have a pinned icon that does something other than open a sidebar? I've been struggling to figure out how to do that in the shipping version of Gutenberg. I think it would be an awesome addition though and makes sense to ensure it's accommodated here.

Reply:
@chrisvanpatten - unfortunately it isn't implemented for modals and menu items that call arbitrary actions. Priorities change too often so I never got back to those APIs. However, technically speaking it should be straightforward to replicate.

As @jasmussen mentioned, we need to fix the issue for the sidebar first - https://github.com/WordPress/gutenberg/issues/14457. In that issue, there is a proposal where the plugin sidebar would have a corresponding menu item attached by default.